backend/src/main/java/com/redhat/cloud/notifications/auth/kessel/KesselAuthorization.java

```diff
@@ -82,6 +86,33 @@ public class KesselAuthorization {
     @Inject
     BackendConfig backendConfig;
 
+    /**
+     * Filters the given list of integrations and leaves only the ones for
+     * which the principal has authorization to view. Useful for
+     * "post-filtering" the integrations once we have fetched them from the
+     * database, and we want to remove the ones that the principal does not
+     * have authorization for.
+     * @param securityContext the security context from which the principal is
+     *                        extracted.
+     * @param endpoints the list of integrations to check.
+     * @return a filtered list of integrations that the principal has
+     * permission to view. The original list is kept untouched to avoid any
+     * issues to avoid "immutable lists cannot be modified" issues.
+     */
+    public List<Endpoint> filterUnauthorizedIntegrations(final SecurityContext securityContext, final List<Endpoint> endpoints) {
+        final List<Endpoint> resultingList = new ArrayList<>();
+
+        for (final Endpoint endpoint : endpoints) {
+            try {
+                this.hasPermissionOnResource(securityContext, IntegrationPermission.VIEW, ResourceType.INTEGRATION, endpoint.getId().toString());
+                resultingList.add(endpoint);
+            } catch (final ForbiddenException ignored) {
+            }
+        }
+
+        return resultingList;
+    }
+
```

Comment:
Even though it is not used right now, I thought that we might want to change our approach in the future depending on the usage we make of the Kessel integration. That way we'll have access to both "pre-filtering" and "post-filtering" methods in case we want to get creative.